settings.php

```diff
@@ -25,23 +25,98 @@
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/enabletasks',
+        new lang_string('settings:enabletasks', 'tool_objectfs'), '', '', '0', '1'));
+
+    $settings->add(new admin_setting_configduration('tool_objectfs/maxtaskruntime',
+        new lang_string('settings:maxtaskruntime', 'tool_objectfs'), '', '', MINSECS));
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/enablelogging',
+        new lang_string('settings:enablelogging', 'tool_objectfs'), '', '', '0', '1'));
+
+
+    $settings->add(new admin_setting_heading('tool_objectfs/filetransfersettings',
+        new lang_string('settings:filetransferheader', 'tool_objectfs'), ''));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/sizethreshold',
+        new lang_string('settings:sizethreshold', 'tool_objectfs'), '', 1024 * 10, PARAM_INT));
+
+    $settings->add(new admin_setting_configtext('tool_objectfs/batchsize',
+        new lang_string('settings:batchsize', 'tool_objectfs'), '', 10000, PARAM_INT));
+
+    $settings->add(new admin_setting_configduration('tool_objectfs/minimumage',
+        new lang_string('settings:minimumage', 'tool_objectfs'), '', '', 7 * DAYSECS));
+
+    $settings->add(new admin_setting_configcheckbox('tool_objectfs/deletelocal',
+        new lang_string('settings:deletelocal', 'tool_objectfs'), '', '', '0', '1'));
+
+    $settings->add(new admin_setting_configduration('tool_objectfs/consistencydelay',
+        new lang_string('settings:consistencydelay', 'tool_objectfs'), '', '', 10 * MINSECS));
+
+
+    $settings->add(new admin_setting_heading('tool_objectfs/storagefilesystemselection',
+        new lang_string('settings:clientselection:header', 'tool_objectfs'), ''));
+
+    $settings->add(new admin_setting_configselect('tool_objectfs/filesystem',
+        new lang_string('settings:clientselection:title', 'tool_objectfs'),
+        new lang_string('settings:clientselection:title_help', 'tool_objectfs'), '',
+        tool_objectfs_get_fs_list()));
+
+
+    $config = get_objectfs_config();
+
+    if (tool_objectfs_filesystem_supports_presigned_urls($config->filesystem)) {
+        $settings->add(new admin_setting_heading('tool_objectfs/presignedurls',
+            new lang_string('settings:presignedurl:header', 'tool_objectfs'), ''));
+
+        $settings->add(new admin_setting_configcheckbox('tool_objectfs/enablepresignedurls',
+            new lang_string('settings:presignedurl:enablepresignedurls', 'tool_objectfs'),
+            new lang_string('settings:presignedurl:enablepresignedurls_help', 'tool_objectfs'), '', '0', '1'));
+
+        $settings->add(new admin_setting_configduration('tool_objectfs/expirationtime',
+            new lang_string('settings:presignedurl:expirationtime', 'tool_objectfs'),
+            new lang_string('settings:presignedurl:expirationtime_help', 'tool_objectfs'), '', 10 * MINSECS));
+
+        $settings->add(new filesize('tool_objectfs/presignedminfilesize',
```

Comment:
not sure about this class `filesize`,  could you post the path to it?

Reply:
Opps, I thought I removed all references to that in one of the last commits. It is 3.8 only and I didn't realise that when I initially added it.

Reply:
Resolved in 654263b